lib/config.js

```diff
@@ -45,45 +45,77 @@ var loadedPlugins = Object.create(null);
 debug = debug("eslint:config");
 
 /**
+ * Determines if a given string represents a filepath or not using the same
+ * conventions as require(), meaning that the first character must be nonalphanumeric
+ * to be considered a file path.
+ * @param {string} filePath The string to check.
+ * @returns {boolean} True if it's a filepath, false if not.
+ * @private
+ */
+function isFilePath(filePath) {
+    return isAbsolutePath(filePath) || !/\w/.test(filePath[0]);
+}
+
+/**
  * Load and parse a JSON config object from a file.
  * @param {string} filePath the path to the JSON config file
  * @returns {Object} the parsed config object (empty object if there was a parse error)
+ * @private
  */
 function loadConfig(filePath) {
     var config = {};
 
     if (filePath) {
-        try {
-            config = yaml.safeLoad(stripComments(fs.readFileSync(filePath, "utf8"))) || {};
-        } catch (e) {
-            debug("Error reading YAML file: " + filePath);
-            e.message = "Cannot read config file: " + filePath + "\nError: " + e.message;
-            throw e;
+
+        if (isFilePath(filePath)) {
+            try {
+                config = yaml.safeLoad(stripComments(fs.readFileSync(filePath, "utf8"))) || {};
+            } catch (e) {
+                debug("Error reading YAML file: " + filePath);
+                e.message = "Cannot read config file: " + filePath + "\nError: " + e.message;
+                throw e;
+            }
+
+        } else {
+
+            // it's a package
+            if (filePath.indexOf("eslint-config-") === -1) {
+                filePath = "eslint-config-" + filePath;
+            }
+
+            config = require(filePath);
```

Comment:
We show a call stack if we can't read a config file, so the behavior is consistent.